Can't add folders via Java GUI in v12 on macOS

After I select the folder, the popup closes and nothing happens, and the logs say:

```
INFO  2022-11-19 20:47:35.180 [AWT-EventQueue-0] Exception in thread "AWT-EventQueue-0" com.google.gson.JsonIOException: Failed making field 'sun.awt.shell.ShellFolder#parent' accessible; either increase its visibility or write a custom TypeAdapter for its declaring type.
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:38)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:287)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:130)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.Gson.getAdapter(Gson.java:546)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:55)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.write(ReflectiveTypeAdapterFactory.java:196)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.write(ReflectiveTypeAdapterFactory.java:366)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.Gson.toJson(Gson.java:825)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.Gson.toJsonTree(Gson.java:695)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.Gson.toJsonTree(Gson.java:672)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.serialize(TreeTypeAdapter.java:172)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at net.pms.configuration.sharedcontent.SharedContentTypeAdapter.serialize(SharedContentTypeAdapter.java:35)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at net.pms.configuration.sharedcontent.SharedContentTypeAdapter.serialize(SharedContentTypeAdapter.java:30)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:88)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:70)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:97)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:61)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.Gson.toJson(Gson.java:825)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.Gson.toJsonTree(Gson.java:695)
INFO  2022-11-19 20:47:35.181 [AWT-EventQueue-0] 	at com.google.gson.Gson.toJsonTree(Gson.java:672)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at net.pms.configuration.sharedcontent.SharedContentConfiguration.getAsJsonArray(SharedContentConfiguration.java:157)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at net.pms.configuration.sharedcontent.SharedContentConfiguration.sendSseApiUpdate(SharedContentConfiguration.java:258)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at net.pms.configuration.sharedcontent.SharedContentConfiguration.updateSharedContent(SharedContentConfiguration.java:133)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at net.pms.newgui.SharedContentTab.lambda$initSharedContentGuiComponents$0(SharedContentTab.java:221)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/javax.swing.DefaultButtonModel.setPressed(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.Component.processMouseEvent(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/javax.swing.JComponent.processMouseEvent(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.Component.processEvent(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.Container.processEvent(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.Component.dispatchEventImpl(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.Container.dispatchEventImpl(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.Component.dispatchEvent(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.Container.dispatchEventImpl(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.Window.dispatchEventImpl(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.Component.dispatchEvent(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
INFO  2022-11-19 20:47:35.182 [AWT-EventQueue-0] 	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field protected sun.awt.shell.ShellFolder sun.awt.shell.ShellFolder.parent accessible: module java.desktop does not "opens sun.awt.shell" to unnamed module @6223440a
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at java.base/java.lang.reflect.Field.setAccessible(Unknown Source)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	at com.google.gson.internal.reflect.ReflectionHelper.makeAccessible(ReflectionHelper.java:35)
INFO  2022-11-19 20:47:35.183 [AWT-EventQueue-0] 	... 59 more
```

It works correctly for me on Windows, just not macOS

--- a/src/main/java/net/pms/newgui/SharedContentTab.java
+++ b/src/main/java/net/pms/newgui/SharedContentTab.java
@@ -29,7 +29,6 @@ import java.awt.FontMetrics;
 import java.awt.Point;
 import java.awt.Rectangle;
 import java.awt.event.*;
-import java.io.File;
 import java.sql.Connection;
 import java.util.ArrayList;
 import java.util.Arrays;

--- a/src/main/java/net/pms/newgui/SharedContentTab.java
+++ b/src/main/java/net/pms/newgui/SharedContentTab.java
@@ -29,6 +29,7 @@ import java.awt.FontMetrics;
 import java.awt.Point;
 import java.awt.Rectangle;
 import java.awt.event.*;
+import java.io.File;
 import java.sql.Connection;
 import java.util.ArrayList;
 import java.util.Arrays;
@@ -214,9 +215,9 @@ public class SharedContentTab implements SharedContentListener {
 			if (returnVal == JFileChooser.APPROVE_OPTION) {
 				int firstSelectedRow = sharedContentList.getSelectedRow();
 				if (firstSelectedRow >= 0) {
-					sharedContentArray.add(firstSelectedRow, new FolderContent(chooser.getSelectedFile()));
+					sharedContentArray.add(firstSelectedRow, new FolderContent(chooser.getSelectedFile().getAbsoluteFile()));
 				} else {
-					sharedContentArray.add(new FolderContent(chooser.getSelectedFile()));
+					sharedContentArray.add(new FolderContent(chooser.getSelectedFile().getAbsoluteFile()));
 				}
 				SharedContentConfiguration.updateSharedContent(sharedContentArray, false);
 			}
@@ -629,7 +630,7 @@ public class SharedContentTab implements SharedContentListener {
 					chooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
 					int returnVal = chooser.showOpenDialog((java.awt.Component) event.getSource());
 					if (returnVal == JFileChooser.APPROVE_OPTION) {
-						folder.setFile(chooser.getSelectedFile());
+						folder.setFile(chooser.getSelectedFile().getAbsoluteFile());
 						SharedContentConfiguration.updateSharedContent(sharedContentArray, true);
 					}
 					return;

